community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/birk/CodeGenerator.scala

```diff
@@ -192,7 +199,13 @@ object CodeGenerator {
   }
 }
 
-case class JavaSymbol(name: String, javaType: String)
+case class JavaSymbol(name: String, javaType: String, materializedSymbol: Option[JavaSymbol] = None) {
```

Comment:
Perhaps we should consider moving `JavaSymbol` and `JavaTypes` to their own file?